src/models/components/previous_mixing.py

```diff
@@ -0,0 +1,20 @@
+import torch
+import torch.nn as nn
+from torch import Tensor
+
+
+class PreviousMixing(nn.Module):
+    def __init__(self, dim: int):
+        super().__init__()
+        self.dim = dim
+        self.mix_factor = nn.Parameter(torch.rand(dim))
+        self.x_last = None
+
+    # (len, dim) -> (len, dim)
+    def forward(self, x: Tensor) -> Tensor:
+        if self.x_last is None:
+            self.x_last = torch.randn(self.dim)
+        x_shift = x.roll(shifts=1, dims=0)
+        x_shift[0] = self.x_last
```

Comment:
@myxyy この実装だとバッチが連続していない場合に意図せず前のバッチの履歴を参照してしまうので、忠実に理論を再現する目的から離れませんかね？それともバッチ学習やらない方針でいきますか？
```suggestion
        x_shift = x.roll(shifts=1, dims=0)
        x_shift[0] = 0.0
```

Reply:
@Geson-anko 
元のRWKVだとバッチを連続させずランダムで学習させてる感じですかね
連続とランダムどっちもやれるようにしたいので隠れ状態をtorch.zeros()でクリアするメソッド生やして
ランダム学習の場合はステップ毎にそれ叩くようにしようと思いますがどうでしょうか

Reply:
@myxyy 
どっちもできるようにしておくのはアリですね！そうしましょう。